src/coreComponents/mesh/NodeManager.hpp

```diff
@@ -431,6 +443,7 @@ class NodeManager : public ObjectManagerBase
   //START_SPHINX_REFPOS
   /// reference position of the nodes
   array2d< real64, nodes::REFERENCE_POSITION_PERM > m_referencePosition;
+  array2d< real32, nodes::REFERENCE_POSITION_PERM > m_referencePosition32;
```

Comment:
I think it would be better to avoid "polluting" the `NodeManager` an keep everything inside the wave equations physics solvers. Hopefully, we will get to the needed refactoring  soon enough for this not to be needed any more.

Reply:
Yes, for the moment, if you need 32 bits positions, please perform the conversion inside your solvers. This shouldn't be a problem and that way we try to keep the architecture as it is while we try to refactor.